### ISSUE TITLE
Support RTL (right-to-left)

### ISSUE THREAD

**Issue by kirill-grouchnikov** [JFormDesigner/FlatLaf#18]
This is a hefty endeavor in general, as core look-and-feels vary greatly in how well they support RTL orientation across the core Swing components. For example, an RTL combobox should have:

1. Items aligned to the right
2. Arrow on the left
3. Popup items aligned to the right
4. Popup scrollbar on the left

Aqua only gets 4 right
Nimbus gets 1, 2, 3 right
Metal gets only 2 right
FlatLaf only gets 2 right at the present moment (but this goes well beyond JComboBox in general)

```java
public class Test {
    public static void main(String[] args) {
        SwingUtilities.invokeLater(new Runnable() {
            @Override
            public void run() {
                try {
                    UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName());
                } catch (Throwable t) {
                    t.printStackTrace(System.err);
                }

                JComboBox comboHebrew = new JComboBox(new Object[] { "\u05e8\u05d0\u05e9\u05d9 1",
                        "\u05e8\u05d0\u05e9\u05d9 2", "\u05e8\u05d0\u05e9\u05d9 3",
                        "\u05e8\u05d0\u05e9\u05d9 4", "\u05e8\u05d0\u05e9\u05d9 5",
                        "\u05e8\u05d0\u05e9\u05d9 6", "\u05e8\u05d0\u05e9\u05d9 7",
                        "\u05e8\u05d0\u05e9\u05d9 8", "\u05e8\u05d0\u05e9\u05d9 9" });
                comboHebrew.setToolTipText("RTL combo");
                comboHebrew.setComponentOrientation(ComponentOrientation.RIGHT_TO_LEFT);
                comboHebrew.setMaximumRowCount(6);

                JComboBox comboRegular = new JComboBox(new Object[] { "item 1",
                        "item 2", "item 3",
                        "item 4", "item 5",
                        "item 6", "item 7",
                        "item 8", "item 9" });
                comboRegular.setToolTipText("LTR combo");
                comboRegular.setComponentOrientation(ComponentOrientation.LEFT_TO_RIGHT);
                comboRegular.setMaximumRowCount(6);

                JButton toFlat = new JButton("Set FlatDark!");
                toFlat.addActionListener(new ActionListener() {
                    @Override
                    public void actionPerformed(ActionEvent e) {
                        SwingUtilities.invokeLater(new Runnable() {
                            @Override
                            public void run() {
                                try {
                                    UIManager.setLookAndFeel("com.formdev.flatlaf.FlatDarkLaf");
                                    for (Window w : Window.getWindows()) {
                                        SwingUtilities.updateComponentTreeUI(w);
                                    }
                                } catch (Throwable t) {
                                    t.printStackTrace(System.err);
                                }
                            }
                        });
                    }
                });

                JButton toNimbus = new JButton("Set Nimbus!");
                toNimbus.addActionListener(new ActionListener() {
                    @Override
                    public void actionPerformed(ActionEvent e) {
                        SwingUtilities.invokeLater(new Runnable() {
                            @Override
                            public void run() {
                                try {
                                    UIManager.setLookAndFeel(new NimbusLookAndFeel());
                                    for (Window w : Window.getWindows()) {
                                        SwingUtilities.updateComponentTreeUI(w);
                                    }
                                } catch (Throwable t) {
                                    t.printStackTrace(System.err);
                                }
                            }
                        });
                    }
                });

                JButton toMetal = new JButton("Set Metal!");
                toMetal.addActionListener(new ActionListener() {
                    @Override
                    public void actionPerformed(ActionEvent e) {
                        SwingUtilities.invokeLater(new Runnable() {
                            @Override
                            public void run() {
                                try {
                                    UIManager.setLookAndFeel(new MetalLookAndFeel());
                                    for (Window w : Window.getWindows()) {
                                        SwingUtilities.updateComponentTreeUI(w);
                                    }
                                } catch (Throwable t) {
                                    t.printStackTrace(System.err);
                                }
                            }
                        });
                    }
                });

                JFrame frame = new JFrame("Combo test");
                frame.setLayout(new BorderLayout());

                JPanel flow = new JPanel(new FlowLayout());
                comboRegular.setPreferredSize(new Dimension(comboRegular.getPreferredSize().width + 50,
                        comboRegular.getPreferredSize().height));
                flow.add(comboRegular);
                comboHebrew.setPreferredSize(new Dimension(comboHebrew.getPreferredSize().width + 50,
                        comboHebrew.getPreferredSize().height));
                flow.add(comboHebrew);
                frame.add(flow, BorderLayout.NORTH);

                JPanel flowButtons = new JPanel(new FlowLayout());
                flowButtons.add(toNimbus);
                flowButtons.add(toMetal);
                flowButtons.add(toFlat);
                frame.add(flowButtons, BorderLayout.SOUTH);

                frame.setSize(400, 400);
                frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
                frame.setLocationRelativeTo(null);
                frame.setVisible(true);
            }
        });
    }
}
```

**Comment by DevCharly** [comment]
thanks, fixed in master branch.

Hope FlatLaf now gets all four right.

have also fixed some rtl issues in Slider and ToolTip

**Comment by JFormDesigner** [comment]
fixed in 0.16
